src/rpc/mining.cpp

```diff
@@ -1346,18 +1350,34 @@ static RPCHelpMan getnewblockhex()
         proposed.m_serialize_type = 2;
     }
 
-    // Any commitment required for non-consensus reasons.
-    // This will be placed in the first coinbase output.
-    CScript data_commitment;
+    // Any commitments required for non-consensus reasons.
+    // These will be placed in the first coinbase outputs.
+    std::set<CScript> data_commitments;
     if (!request.params[2].isNull()) {
-        std::vector<unsigned char> data_bytes = ParseHex(request.params[2].get_str());
-        data_commitment = CScript() << OP_RETURN << data_bytes;
+        UniValue commitments(UniValue::VARR);
+
+        // backwards compatibility: attempt to parse as a string first
+        if (request.params[2].isStr()) {
+            UniValue hex = request.params[2].get_str();
+            commitments.push_back(hex);
+        } else {
+            commitments = request.params[2].get_array();
+        }
+
+        for (unsigned int i = 0; i < commitments.size(); i++) {
+            std::vector<unsigned char> data_bytes = ParseHex(commitments[i].get_str());
+            CScript data_commitment = CScript() << OP_RETURN << data_bytes;
+            auto res = data_commitments.insert(data_commitment);
+            if (!res.second) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Duplicate commitment");
+            }
```

Comment:
In 110087764022671d699891592e74fe89a4adae37:

Also not going to hold up the PR for this, but noting in case people run into this, so they'll see that we at least thought about it. Why do we have this restriction?

Reply:
I wasn’t sure it made sense to allow duplicate commitments in the same call?